src/core/jit/store.rs

```diff
@@ -1,11 +1,19 @@
+use std::iter::repeat_with;
+
 use crate::core::jit::model::FieldId;
 
+#[derive(Debug, Clone)]
+pub struct MultipleDataPath {
```

Comment:
Add some documentation about why is this needed.

Reply:
added docs

Reply:
Let's simplify this for now by using a `HashMap<FieldId, Data<A>>` so that we don't need to keep a track of the size everytime we insert.